hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicConfigReplicationOperation.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.dynamicconfig;
+
+import com.hazelcast.config.ConfigDataSerializerHook;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import java.io.IOException;
+
+public class DynamicConfigReplicationOperation extends AbstractDynamicConfigOperation {
+
+    private IdentifiedDataSerializable[] configs;
+    private boolean failWhenNotEquals;
+
+    public DynamicConfigReplicationOperation(IdentifiedDataSerializable[] configs, boolean failWhenNotEquals) {
+        this.configs = configs;
+        this.failWhenNotEquals = failWhenNotEquals;
+    }
+
+    public DynamicConfigReplicationOperation() {
+
+    }
+
+    @Override
+    public void run() throws Exception {
+        ClusterWideConfigurationService service = getService();
+        for (IdentifiedDataSerializable config : configs) {
+            service.registerConfigLocally(config, failWhenNotEquals);
```

Comment:
1. I don't think failing migration is a good option, unless this error is recoverable eventually. Since failed migration will be retried soon.

2. I'm not sure how overwriting helps, since it doesn't solve the problem, another migration from another member can overwrite back previous configs.

3. I think logging is only choice here. Also consider that, migration failures are not visible by users directly, they will be logged too.